python/cuml/explainer/sampling.py

```diff
@@ -65,24 +61,19 @@ def kmeans_sampling(X, k, round_values=True, detailed=False, random_state=0):
                         dtypes: cuDF DataFrame, cuDF Series, cupy, numba,\
                         numpy, pandas DataFrame, pandas Series")
 
-    if output_dtype == cudf.DataFrame:
+    if "DataFrame" in str(output_dtype):
         group_names = X.columns
         X = X.values
-    elif output_dtype == cudf.Series:
+        if "pd" in str(output_dtype):
+            X = cp.array(X.values)
```

Comment:
you can avoid a few lines and the second conditional by doing: 

```
X = cp.array(X.values, copy=False) 
```

the same for both cudf and pandas objects, similar in series